main/src/main/scala/sbt/Load.scala

```diff
@@ -74,8 +74,21 @@ object Load {
   def buildGlobalSettings(base: File, files: Seq[File], config: sbt.LoadBuildConfiguration): ClassLoader => Seq[Setting[_]] =
     {
       val eval = mkEval(data(config.globalPluginClasspath), base, defaultEvalOptions)
-      val imports = BuildUtil.baseImports ++ BuildUtil.importAllRoot(config.globalPluginNames)
+
       loader => {
+        val pg = PluginData(config.globalPluginClasspath, None, None)
+        val discovery = PluginDiscovery.discoverAll(pg, loader)
+
+        val detectedPlugins =
```

Comment:
I don't think I can, since PluginDiscovery requires a ClassLoader, and I don't think a ClassLoader is available in LoadBuildConfiguration.